bluez-qt/bluetoothdevicemodel.cpp

```diff
@@ -176,14 +176,17 @@ void BluetoothDevicesModel::defaultAdapterChanged(QDBusObjectPath path)
 
 	if (adapter)
 	{
-		beginRemoveRows(QModelIndex(), 0, m_devices.size()-1);
-		foreach(BluetoothDevice* device, m_devices)
+		if (m_devices.size())
 		{
-			updateConnected(device->connected());
-			device->deleteLater();
+			beginRemoveRows(QModelIndex(), 0, m_devices.size()-1);
+			foreach(BluetoothDevice* device, m_devices)
+			{
+				updateConnected(device->connected());
```

Comment:
What's the purpose of this updateConnected() call if we are about to delete the device?